int32 objNum
objInfo[] data
coeff[] coeff